src/main/java/com/cinema/security/CustomUserDetailService.java

```diff
@@ -18,15 +17,18 @@ public CustomUserDetailService(UserService userService) {
     }
 
     @Override
-    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
-        User user = userService.findByEmail(email).orElseThrow();
-        UserBuilder builder = org.springframework.security.core
-                .userdetails.User.withUsername(email);
-        builder.password(user.getPassword());
-        builder.roles(user.getRoles().stream()
-                .map(r -> r.getRoleName().toString())
-                .collect(Collectors.toList())
-                .toArray(String[]::new));
-        return builder.build();
+    public UserDetails loadUserByUsername(String email) {
+        try {
+            User user = userService.findByEmail(email).get();
```

Comment:
does it work?
```suggestion
            User user = userService.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException("User with email " + email + "not found", e));
```

Reply:
No - it does come to this point, but final message I got is not the same  - I got error 401 (Unauthorised) and message "The request has not been applied because it lacks valid authentication credentials for the target resource"